base_layer/core/Cargo.toml

```diff
@@ -94,7 +94,11 @@ tiny-keccak = { package = "tari-tiny-keccak", version = "2.0.2", features = [
     "keccak",
 ] }
 dirs-next = "1.0.2"
-hickory-client = { version = "0.25.0-alpha.2", features = ["dns-over-rustls", "dnssec-openssl"], optional = true }
+hickory-client = { version = "=0.25.0-alpha.2", features = [
+    "dns-over-rustls",
+    "dnssec-openssl",
+], optional = true }
+hickory-proto = { version = "=0.25.0-alpha.2" }
```

Comment:
why is this added in here?

Reply:
When I've locked` hickory-client` to  `0.25.0-alpha.2` for some reason this `hickory-proto` still tried to use `alpha 4` version of itself which broke the build so I added entry for this package with also locked version to `0.25.0-alpha.2`

Reply:
you are adding a package thats not used by tari. You can see its complaining about it in machete

Reply:
This is needed because hickory pushed a new version that doesn't compile, so you have to pin it like this.